src/query/service/src/interpreters/interpreter_table_create.rs

```diff
@@ -82,20 +82,17 @@ impl Interpreter for CreateTableInterpreter {
         let quota = quota_api.get_quota(MatchSeq::GE(0)).await?.data;
         let engine = self.plan.engine;
         let catalog = self.ctx.get_catalog(self.plan.catalog.as_str())?;
-        let tables = catalog
-            .list_tables(&self.plan.tenant, &self.plan.database)
-            .await?;
-        if quota.max_tables_per_database != 0
-            && tables.len() >= quota.max_tables_per_database as usize
-        {
-            return Err(ErrorCode::TenantQuotaExceeded(format!(
-                "Max tables per database quota exceeded: {}",
-                quota.max_tables_per_database
-            )));
-        };
-        let name_not_duplicate = tables
-            .iter()
-            .all(|table| table.name() != self.plan.table.as_str());
+        if quota.max_tables_per_database != 0 {
```

Comment:
Suggest adding some comments here, to avoid someone modify in the future :)

Reply:
How about this? https://github.com/datafuselabs/databend/pull/11427#discussion_r1192196915

I think this check is no needed.